apps/portal/src/components/common/ProductsSection.js

```diff
@@ -978,6 +978,11 @@ function ProductsSection({onPlanSelect, products, type = null, handleChooseSignu
         className += ' gh-portal-upgrade-product';
     }
 
+    // If site doesn't have paid products then don't return an empty container
+    if (hasOnlyFreeProduct({site})) {
```

Comment:
https://github.com/TryGhost/Ghost/blob/8e0afbd18ebc204e3968c1da44a31037a9fe9e5d/apps/portal/src/components/common/ProductsSection.js#L959-L961

We are already checking whether we have paid products or not (although with a different condition).
Should the above condition be `true`, this line will never be reached.

Reply:
Also you can always use [hasOnlyFree](https://github.com/TryGhost/Ghost/blob/8e0afbd18ebc204e3968c1da44a31037a9fe9e5d/apps/portal/src/components/common/ProductsSection.js#L667) in place of `hasOnlyFreeProduct({site})` if you still want to use it.

Reply:
I think this is a different one as it checks which cadences are checked, while the other checks which products are checked in Portal settings. However my change could be combined with this condition like this:

```
if ((!portalPlans.includes('monthly') && !portalPlans.includes('yearly')) || hasOnlyFreeProduct({site})) {
        return null;
    }
    ```
    
 Wdyt?